### PR TITLE
Product Task complete

### PR DESCRIPTION
**Pull request by muhammadharoon63**
- made changes according to feedback in Products CRUD,Pagination,Export CSV
-Added Users pagination with page
-Added Search and sort for Products
-Added Search and sort for Category

### CODE REVIEW

File: app/controllers/admin/categories_controller.rb [hunk 1]
```diff
@@ -3,11 +3,19 @@ class CategoriesController < Admin::BaseController
...
 
     def index
-      @pagy, @category = pagy(Category.all, items: 5)
+      @pagy, @categories = pagy(Category.all, items: 5)
       respond_to do |format|
         format.html
         format.csv { send_data Category.all.to_csv, filename: "categories-#{Date.today}.csv" }
       end
+
+      if params[:search].present?
```
Comment (adam-wali) [inline app/controllers/admin/categories_controller.rb]: Can we refactor this in such a way where we always call the search method and it early returns the all categories if the search param is blank? In that way first, we can avoid conditional here. Secondly order line will be used once as well.
Lastly, the order line is quite long. Make a conditional block for that.

File: app/controllers/admin/products_controller.rb [hunk 2]
```diff
@@ -1,9 +1,16 @@
 module Admin
   class ProductsController < BaseController
-    before_action :set_product, only: [:show, :edit, :update, :destroy]
+    before_action :set_product, only: %i[show edit update destroy]
 
     def index
       @pagys, @products = pagy(Product.all, items: 5)
+      if params[:search].present?
```
Comment (adam-wali) [inline app/controllers/admin/products_controller.rb]: Same as above

File: app/controllers/admin/users_controller.rb [hunk 3]
```diff
@@ -1,16 +1,16 @@
...
-    include Orderable
+
     before_action :set_user, only: %i[show edit destroy]
-    helper_method :sort_column, :sort_direction
+
 
     def index
-      @users = User.all
+      @pagyz, @users = pagy(User.client, items: 5)
       if params[:search].present?
```
Comment (adam-wali) [inline app/controllers/admin/users_controller.rb]: Same as above

File: app/controllers/concerns/orderable.rb [hunk 4]
```diff
@@ -2,12 +2,12 @@ module Orderable
   extend ActiveSupport::Concern
 
   included do
-    def sort_column(user)
-      user.column_names.include?(params[:sort]) ? params[:sort] : 'username'
+    def sort_column(temp)
```
Comment (adam-wali) [inline app/controllers/concerns/orderable.rb]: Use better variable names unlike `temp`

File: app/models/category.rb [hunk 5]
```diff
@@ -12,4 +12,9 @@ def self.to_csv
       end
     end
   end
+
+  def self.search(search)
+    where('categories.id LIKE ? || categories.name LIKE ?', "%#{search}%", "%#{search}%")
```
Comment (adam-wali) [inline app/models/category.rb]: Should not it be `OR` in-place of `||`

File: app/models/product.rb [hunk 6]
```diff
@@ -13,6 +13,10 @@ def self.to_csv
     end
   end
 
+  def self.search(search)
+    where('products.id LIKE ? || products.title LIKE ? || products.status LIKE ?', "%#{search}%", "%#{search}%", "%#{search}%")
```
Comment (adam-wali) [inline app/models/product.rb]: Same as above

File: app/models/category.rb [hunk 7]
```diff
@@ -14,7 +14,8 @@ def self.to_csv
   end
 
   def self.search(search)
-    where('categories.id LIKE ? || categories.name LIKE ?', "%#{search}%", "%#{search}%")
+    return all if search.blank?
```
Comment (fred-azhar) [inline app/models/category.rb:9]: We should always add a new line after early return. Please do so

File: app/models/product.rb [hunk 8]
```diff
@@ -14,7 +14,8 @@ def self.to_csv
   end
 
   def self.search(search)
-    where('products.id LIKE ? || products.title LIKE ? || products.status LIKE ?', "%#{search}%", "%#{search}%", "%#{search}%")
+    return all if search.blank?
```
Comment (fred-azhar) [inline app/models/product.rb:12]: Same as above

File: app/models/user.rb [hunk 9]
```diff
@@ -11,7 +11,8 @@ class User < ApplicationRecord
   attr_writer :login
 
   def self.search(search)
-    where('users.username LIKE ? || users.email LIKE ? || users.id LIKE ?', "%#{search}%", "%#{search}%", "%#{search}%")
+    return all if search.blank?
```
Comment (fred-azhar) [inline app/models/user.rb:14]: Same as above

**Pull request merged**